src/main/java/com/minecolonies/rankup/util/RankingUtils.java

```diff
@@ -78,42 +67,42 @@ public static void balanceCheck(final Player player, final Rankup plugin)
     {
-        final GroupsConfig groupsConfig = (GroupsConfig) plugin.getAllConfigs().get(GroupsConfig.class);
-        final AccountConfigData playerData = (AccountConfigData) plugin.getAllConfigs().get(AccountConfigData.class);
-
-        final List<String> playerGroups = CoreModule.perms.getPlayerGroupIds(player);
+        final GroupsConfig groupsConfig = plugin.configUtils.getGroupsConfig(player);
+        final List<String> playerGroups = plugin.perms.getPlayerGroupIds(player);
 
         //Check if player is in disabled group.
-        for (final Subject subject : CoreModule.perms.getDisabledGroups())
+        for (final String subject : plugin.perms.getDisabledGroups())
         {
-            if (playerGroups.contains(subject.getIdentifier()))
+            if (playerGroups.contains(subject))
             {
                 return; //IF the player is withing a disabled group, stop here.
             }
         }
 
-        for (final String group : CoreModule.perms.getPlayerGroupIds(player))
+        for (final String group : plugin.perms.getPlayerGroupIds(player))
         {
-            final Integer time = playerData.playerData.get(player.getUniqueId()).timePlayed;
-
-            if (groupsConfig.groups.get(group).timingRankDown && groupsConfig.groups.get(group).timingTime > time)
+            if (groupsConfig.groups.containsKey(group))
             {
-                rankDown(player, plugin);
+                final Integer time = plugin.accUtils.getPlayerTime(player.getUniqueId());
+
+                if (groupsConfig.groups.get(group).timingRankDown && groupsConfig.groups.get(group).timingTime > time)
+                {
+                    rankDown(player, plugin);
+                }
             }
         }
     }
 
     private static void rankDown(final Player player, final Rankup plugin)
     {
-        final GroupsConfig groupsConfig = (GroupsConfig) plugin.getAllConfigs().get(GroupsConfig.class);
-        
-        final String currentGroup = CoreModule.perms.getPlayerHighestRankingGroup(player);
-        final String previousGroup = CoreModule.perms.getPreviousGroup(groupsConfig.groups.get(currentGroup).rank);
+        final GroupsConfig groupsConfig = plugin.configUtils.getGroupsConfig(player);
 
-        final CoreConfig coreConfig = plugin.getConfigAdapter(CoreModule.ID, CoreConfigAdapter.class).get().getNodeOrDefault();
+        final String currentGroup = plugin.perms.getPlayerHighestRankingGroup(player);
+        final String previousGroup = plugin.perms.getPreviousGroup(player, groupsConfig.groups.get(currentGroup).rank);
 
-        final String remCmd = coreConfig.rankdownCommand;
+        CoreConfig coreConfig = plugin.configUtils.getCoreConfig();
 
+        final String remCmd = coreConfig.rankdownCommand;
         final String finalRemCmd = remCmd.replace("{player}", player.getName())
                                      .replace("{group}", previousGroup);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "{group}" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)